### DISCUSSION TITLE
free-tools/code-to-image-converter

### DISCUSSION THREAD

**Discussion by giscus** [EinGuterWaran/boilerplatehub-comments#2] [Announcements]
# free-tools/code-to-image-converter

Discover your ideal tech stack with our interactive recommender tool.

https://boilerplatehub.com/code-to-image-converter

**Comment by w68146046-png** [comment]
# Combine all floors into one printable PDF, with color-coded, labeled, and detailed 2D floor maps.
# Each floorplan is enhanced with colors, furniture symbols, and basic dimension labels.

from reportlab.lib.pagesizes import A3, landscape
from reportlab.pdfgen import canvas
from reportlab.lib.utils import ImageReader

pdf_path = "/mnt/data/Mini_Hospital_All_Floors.pdf"
c = canvas.Canvas(pdf_path, pagesize=landscape(A3))
width, height = landscape(A3)

floor_images = [
    ("/mnt/data/floor_maps/ground_floor.png", "Ground Floor — Reception / Waiting & Pharmacy"),
    ("/mnt/data/floor_maps/first_floor.png", "First Floor — Doctor Office + Emergency Operation Room"),
    ("/mnt/data/floor_maps/second_floor.png", "Second Floor — 2 Doctor Offices + Waiting & Support"),
    ("/mnt/data/floor_maps/third_floor.png", "Third Floor — same as Second Floor"),
    ("/mnt/data/floor_maps/fourth_floor.png", "Fourth Floor — same as Second Floor")
]

# Layout enhancements: titles, color legend, image placements
for img_path, title in floor_images:
    c.setFont("Helvetica-Bold", 18)
    c.drawCentredString(width / 2, height - 50, "Mini Hospital — 1361.25 sq ft (45' × 30.25')")
    c.setFont("Helvetica", 14)
    c.drawCentredString(width / 2, height - 80, title)

    # Draw the floor image
    img = ImageReader(img_path)
    img_w, img_h = img.getSize()
    scale_factor = (width - 200) / img_w
    scaled_h = img_h * scale_factor
    y_pos = (height - scaled_h) / 2 - 20
    c.drawImage(img, 100, y_pos, width=width - 200, height=scaled_h, mask='auto')

    # Draw legend
    c.setFont("Helvetica", 12)
    c.drawString(100, 60, "Color Legend:")
    c.setFillColorRGB(0.2, 0.4, 1.0)
    c.rect(200, 50, 20, 10, fill=True, stroke=False)
    c.setFillColorRGB(0, 0, 0)
    c.drawString(230, 50, "Reception / Waiting Area")

    c.setFillColorRGB(0.4, 0.8, 0.4)
    c.rect(430, 50, 20, 10, fill=True, stroke=False)
    c.setFillColorRGB(0, 0, 0)
    c.drawString(460, 50, "Pharmacy / Support Rooms")

    c.setFillColorRGB(1.0, 0.3, 0.3)
    c.rect(660, 50, 20, 10, fill=True, stroke=False)
    c.setFillColorRGB(0, 0, 0)
    c.drawString(690, 50, "Emergency Operation Room")

    c.setFillColorRGB(0.2, 0.2, 0.8)
    c.rect(900, 50, 20, 10, fill=True, stroke=False)
    c.setFillColorRGB(0, 0, 0)
    c.drawString(930, 50, "Doctor Offices / Consultation")

    c.showPage()

c.save()
pdf_path
